### PR TITLE
Rewrite history, it's just like the History Channel.

### PR DESCRIPTION
**Pull request by leogrim**
@ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala [hunk 1]
```diff
@@ -252,6 +257,92 @@ class AdminPaymentsController @Inject() (
     Ok(views.html.admin.paymentsDashboard(dashboard))
   }
 
+  // todo(Léo): REMOVE THIS IS EVIL
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:260]: :grin:

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala [hunk 2]
```diff
@@ -252,6 +257,92 @@ class AdminPaymentsController @Inject() (
...
+  def resetAllAccounts(doIt: Boolean) = AdminUserAction { implicit request =>
+    if (doIt) {
+      SafeFuture {
+        val activeOrgIds = db.readOnlyMaster { implicit session =>
+          orgRepo.allActive.map(_.id.get)
+        }
+        activeOrgIds.map(doResetAccount)
+      }
+      Ok("Hold your breath.")
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:269]: How much do you care about this? Do you want to do a chunked response?

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala [hunk 3]
```diff
@@ -252,6 +257,92 @@ class AdminPaymentsController @Inject() (
...
+          activeUsers = 0
+        ))
+      } getOrElse {
+        throw new LockedAccountException(orgId)
+      }
+
+      // Deactivate all existing events
+      accountEventRepo.getAllByAccount(account.id.get).foreach { event =>
+        accountEventRepo.save(event.withState(AccountEventStates.INACTIVE))
+      }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:306]: Are there any unique indices that are going to be sad?

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala [hunk 4]
```diff
@@ -252,6 +257,92 @@ class AdminPaymentsController @Inject() (
...
+
+      // Register all members
+      memberships.toSeq.sortBy(_.createdAt).foreach { membership =>
+        val event = planCommander.registerNewUser(orgId, membership.userId, membership.role, ActionAttribution(Some(organization.ownerId), None))
+        accountEventRepo.save(event.copy(eventTime = membership.createdAt))
+      }
+
+      accountEventRepo.getAllByAccount(account.id.get)
+    }
+  }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:344]: :+1:

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala (hunk 2 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:269]: Nope

File: kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala (hunk 3 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:306]: Not if I just deactivate them. Then I'll delete inactive rows.

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/admin/AdminPaymentsController.scala:306]: I meant when you start creating new events.

**Comment by ryanpbrewster** [comment]
Does this compile now?

**Pull request merged**